tests/store/artifact/test_cloud_artifact_repo.py

```diff
@@ -16,3 +24,107 @@ def test_chunk_size_validation_failure():
         _validate_chunk_size_aws(5 * 1024**2 - 1)
     with pytest.raises(MlflowException, match="Multipart chunk size"):
         _validate_chunk_size_aws(5 * 1024**3 + 1)
+
+
+@pytest.mark.parametrize(
+    "future_result, expected_call_count",
+    [
+        (None, 2),  # Simulate where creds are expired, but successfully download after refresh
+        (
+            Exception("fake_exception"),
+            4,
+        ),  # Simulate where there is a download failure and retries are exhausted
+    ],
+)
+def test__parallelized_download_from_cloud(monkeypatch, future_result, expected_call_count):
```

Comment:
How is refresh_credential tested? Could we add a test and mock a credential usage